### PR TITLE
implement user update profile feature

### PR DESCRIPTION
**Pull request by danoseun**
#### What does this PR do?
This PR avails the user of this product to be able to update his/her user profile.
#### Description of Task to be completed
- When user logs in/signs up, a token is supplied
- The token is checked for validity and the necessary details are picked from there.
- The user is found in the database and then the user can proceed to update the chosen records.
#### How should this be manually tested?
- Clone the repo, run `npm install` and `npm test`
- On postman, you can use `localhost:<port>/profile` for a put or get request
- Supply **any of** or **all of** `first_name`, last_name`, `address`, `preferred_lang`, `preferred_currency`, `gender` as request body.
#### What are the relevant pivotal tracker stories?
- 167730636

### CODE REVIEW

File: src/routes/api/authRoutes.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const { sendPasswordResetEmail, receiveNewPassword } = resetPassword;
 
 
 const route = Router();
-const { signup, fbgooglesignup, login, logout} = authControllers;
+const { signup, fbgooglesignup, login, logout, getUserProfile, updateProfile } = authControllers;
```
Comment (chuksjoe) [inline src/routes/api/authRoutes.js]: why not use `import destructuring`

File: src/services/authServices.js [hunk 2]
```diff
@@ -57,3 +58,32 @@ export const logoutService = async (token) => {
     throw err;
   }
 };
+
+export const getUserProfileService = async(id) => {
```
Comment (chuksjoe) [inline src/services/authServices.js]: this function does the same thing as the `findUserById`. I think it's redundant.

**Review by chuksjoe — CHANGES_REQUESTED** [review]
You can attend to Hound flags and other inline comments I dropped.

File: src/routes/api/authRoutes.js (hunk 1 above)
Comment (danoseun) [reply src/routes/api/authRoutes.js]: I just followed the structure I met on ground.

**Comment by Kaytbode** [comment]
@danoseun  Good job mate..

please attend to hound

File: src/services/authServices.js (hunk 2 above)
Comment (danoseun) [reply src/services/authServices.js]: There is a `delete` line in this one

**Review by captainamiedi — COMMENTED** [review]
nice one bro. i think your test coverage is poor. can you write more test especially at the update user profile

**Review by captainamiedi — COMMENTED** [review]
i think you should add some validations at the update as well.

**Pull request merged**